api/src/main/java/org/openmrs/api/CohortService.java

```diff
@@ -140,13 +140,16 @@
 	 *
 	 * @since 2.1
 	 * @param patient patient used to find the cohorts
-	 * @return Cohorts that contain the given patient
+	 * @param includeVoidedCohorts voided true/false whether or not to include voided cohorts
+	 * @param includeVoidedMemberships voided true/false whether or not to include voided memberships
+	 *@param asOfDate  @return Cohorts that contain the given patient
 	 * @throws APIException
 	 * @should not return voided cohorts
 	 * @should return cohorts that have given patient
 	 */
 	@Authorized( { PrivilegeConstants.GET_PATIENT_COHORTS })
-	public List<Cohort> getCohortsContainingPatient(Patient patient, Boolean voided) throws APIException;
+	public List<Cohort> getCohortsContainingPatient(Patient patient, Boolean includeVoidedCohorts,
+													Boolean includeVoidedMemberships, Date asOfDate) throws APIException;
```

Comment:
You shouldn't have the includeVoidedMemberships on this method, instead you should add another  `getMemberships(boolean includeVoided)` method to the Cohort class